datalad/cmd.py

```diff
@@ -104,26 +156,13 @@ def run(self, cmd, log_stdout=True, log_stderr=True):
             proc = subprocess.Popen(cmd, stdout=outputstream, stderr=errstream, shell=True)
             # shell=True allows for piping, multiple commands, etc., but that implies to not use shlex.split()
             # and is considered to be a security hazard. So be careful with input.
-            # Alternatively we would have to parse `cmd` and create multiple subprocesses.
-
-            while proc.poll() is None:
-                if log_stdout:
-                    line = proc.stdout.readline()
-                    if line != '':
-                        self.log("stdout| " + line.rstrip('\n'))
-                        # TODO: what level to log at? was: level=5
-                        # Changes on that should be properly adapted in test.cmd.test_runner_log_stdout()
-                else:
-                    pass
-
-                if log_stderr:
-                    line = proc.stderr.readline()
-                    if line != '':
-                        self.log("stderr| " + line.rstrip('\n'), level=logging.ERROR)
-                        # TODO: what's the proper log level here?
-                        # Changes on that should be properly adapted in test.cmd.test_runner_log_stderr()
-                else:
-                    pass
+            # Alternatively we would have to parse `cmd` and create multiple
+            # subprocesses.
```

Comment:
yeah, to don't deal with all this, ans since Popen has cwd option (so we don't need cd dir &&) we could provide arguments properly split from the beginning, thus avoiding any explicit need to escape shell.
If we are to resort to splitting -- we could use   shlex standard library: http://pymotw.com/2/shlex/